components/exercises/chapter-4-oscar/Utils.tsx

```diff
@@ -70,3 +72,229 @@ export function reverseArrayInPlace(arr: any[]): undefined {
+      food = 'Goulash';
+      break;
+  }
+  return <p>{food}</p>;
+}
+
+/**
+ * Represents a list item.
+ * @param item - The item to be rendered.
+ * @returns The rendered list item.
+ */
+export function ListItem({ item }: { item: ListItemProp }) {
+  return (
+    <li>
+      <p className='text-gray-800'>{item.value}</p>
+      {item.rest === null ? (
+        <div className='mb-2'>
+          <div className='cursor-pointer rounded-lg bg-red-600 p-3 shadow'>
+            <p className='font-semibold'>No more cities to visit</p>
+          </div>
+        </div>
+      ) : (
+        <details className='mb-2'>
+          <summary className='cursor-pointer rounded-lg bg-yellow-600 p-3 shadow'>
+            <span className='font-semibold'>
+              {item.rest && Object.getOwnPropertyNames(item?.rest)[1]}
+            </span>
+          </summary>
+          <div className='bg-white p-4'>
+            <ListItem item={item.rest} />
+          </div>
+        </details>
+      )}
+    </li>
+  );
+}
+
+/**
+ * Represents the properties of a week item.
+ */
+export type WeekItem = {
+  name: string;
+  description: string;
+  price: number;
+};
+
+/**
+ * Represents a list of item to work as parameters for DeepCoparison component.
+ */
+export const lastWeekList: WeekItem[] = [
+  {
+    name: '🥚 Eggs',
+    description: '6x Large class A free range eggs',
+    price: 2.5,
+  },
+  {
+    name: '🐓 Chicken',
+    description: '650gr Fresh Class A skinless chicken breast fillet portions.',
+    price: 4.7,
+  },
+  {
+    name: '🍚 Basmati rice',
+    description:
+      '1kg Aromatic & fluffy A long, slender grain grown in the foothills of the Himalayas',
+    price: 2.5,
+  },
+  {
+    name: '🥔 Potatoes',
+    description: '1kg White potatoes',
+    price: 1.5,
+  },
+];
+/**
+ * Represents a list of item to work as parameters for DeepCoparison component.
+ */
+
+export const currentWeekList: WeekItem[] = [
+  {
+    name: '🥚 Eggs',
+    description: '6x Large class A free range eggs',
+    price: 2.5,
+  },
+  {
+    name: '🐓 Chicken',
+    description: '650gr Fresh Class A skinless chicken breast fillet portions.',
+    price: 4.7,
+  },
+  {
+    name: '🍚 Basmati rice',
+    description:
+      '1kg Aromatic & fluffy A long, slender grain grown in the foothills of the Himalayas',
+    price: 2.5,
+  },
+  {
+    name: '🥔 Potatoes',
+    description: '1kg White potatoes',
+    price: 1.5,
+  },
+];
+
+/**
+ * Checks if two values are deeply equal.
+ * @param a - The first value to compare.
+ * @param b - The second value to compare.
+ * @returns True if the values are deeply equal, false otherwise.
+ */
+
+export function checkEqual(a: any, b: any): boolean {
+  if (a === b) return true;
+
+  // This check ensures that if either a or b is not an object (including arrays), they are not considered equal
+  if (a == null || typeof a != 'object' || b == null || typeof b != 'object')
+    return false;
+
+  let keysA = Object.keys(a),
```

Comment:
if you are not going to reassign keysA or keysB to a different value, use `const` instead of `let`

Reply:
refactored, thank you